tgui/packages/tgui/interfaces/CharacterPreferences/SavesTab.tsx

```diff
@@ -0,0 +1,51 @@
+import { Fragment } from 'inferno';
+import { useBackend } from '../../backend';
+import { Box, Button, Divider, LabeledList, Section } from '../../components';
+import { CharacterPreferencesData } from './type';
+
+export const SavesTab = (_props, context) => {
+  const { act, data } = useBackend<CharacterPreferencesData>(context);
+
+  return (
+    <Section title="Cloud Saves">
+      {data.cloudSaves ? (
+        <Fragment>
+          {data.cloudSaves.map((save, index) => (
+            <Fragment key={save.name}>
+              <Cloudsave name={save.name} index={index} />
+              <Divider />
+            </Fragment>
+          ))}
+          <Box mt="5px">
+            <Button onClick={() => act('cloud-new')}>Create new save</Button>
+          </Box>
+        </Fragment>
+      ) : (
+        <Box italic color="label">
+          Cloud saves could not be loaded.
+        </Box>
+      )}
+    </Section>
+  );
+};
+
+const Cloudsave: (props: { name: string; index: number }, context: any) => JSX.Element = ({ name, index }, context) => {
```

Comment:
Syntax here seems weird. Is this different from:
```
const Cloudsave = (props, context) => {
  const { name, index } = props;
  const { act, data } = useBackend<CharacterPreferencesData>(context);
  return (
   // all of the other stuff
  );
};

Reply:
Inlining types is pretty bad, you should extract it into a separate type, e.g. `CloudsaveFC`.